test/test-network/systemd-networkd-tests.py

```diff
@@ -2311,5 +2343,26 @@ def test_dhcp_client_route_remove_on_renew(self):
         self.assertRegex(output, f'192.168.5.1 proto dhcp scope link src {address2} metric 1024')
 
 if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build-dir', help='Path to build dir', dest='build_dir')
+    parser.add_argument('--networkd', help='Path to systemd-networkd', dest='networkd_bin', default=networkd_bin)
+    parser.add_argument('--wait-online', help='Path to systemd-networkd-wait-online', dest='wait_online_bin', default=wait_online_bin)
+    parser.add_argument('--networkctl', help='Path to networkctl', dest='networkctl_bin', default=networkctl_bin)
+    parser.add_argument('--valgrind', help='Enable valgrind', dest='use_valgrind', type=bool, nargs='?', const=True, default=use_valgrind)
+    parser.add_argument('--log-debug', help='Generate debugging logs', dest='enable_debug', type=bool, nargs='?', const=True, default=enable_debug)
+    ns, args = parser.parse_known_args(namespace=unittest)
+
+    if ns.build_dir:
+        networkd_bin = os.path.join(ns.build_dir, 'systemd-networkd')
+        wait_online_bin = os.path.join(ns.build_dir, 'systemd-networkd-wait-online')
+        networkctl_bin = os.path.join(ns.build_dir, 'networkctl')
+    else:
+        networkd_bin = ns.networkd_bin
+        wait_online_bin = ns.wait_online_bin
+        networkctl_bin = ns.networkctl_bin
+    use_valgrind = ns.use_valgrind
```

Comment:
I'm pretty sure this is your favorite option :-) but while reading the code for the first time I expected it to cause everything to be run under Valgrind including `networkctl` and `wait-online`. Are you planning to add something like `--REALLY-USE-VALGRIND-EVERYWHERE` later? :-)

Just to be clear, I have no problem with the option and it looks good to me as long as you're OK with it :-)

Reply:
OK. Now all networkd related commands run under valgrind or with `$SYSTEMD_LOG_LEVEL=debug` when corresponding options are specified.